src/timesync/timesyncd-server.c

```diff
@@ -175,3 +178,39 @@ void server_name_flush_addresses(ServerName *n) {
         while (n->addresses)
                 server_address_free(n->addresses);
 }
+
+int server_name_parse_port(ServerName *n) {
+        const char *sqr = "]", *col = ":";
+        char *ret_sqr, *ret_col, *last_col;
+
+        ret_sqr = strrchr(n->string, *sqr);
+        ret_col = strchr(n->string, *col);
+        last_col = strrchr(n->string, *col);
+
+        if (ret_sqr == NULL && ret_col == NULL) { // server.domain or I.P.v.4
+                return 0;
+        } else if (ret_sqr == NULL && strlen(ret_col) == strlen(last_col)) { // has no ']' and exactly one ":"
```

Comment:
as per coding style we do not use {} on single line if blocks.

as per coding style we do not compare explicitly with == NULL, but rely on C's downgrade-to-bool feature for pointer non-null checks

Reply:
Noted. I will address this.